catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/query/GeoNamesQueryLuceneDirectoryIndex.java

```diff
@@ -30,14 +30,27 @@
 import org.codice.ddf.spatial.geocoding.context.NearbyLocation;
 import org.codice.ddf.spatial.geocoding.index.GeoNamesLuceneIndexer;
 import org.locationtech.spatial4j.context.SpatialContext;
+import org.locationtech.spatial4j.context.jts.JtsSpatialContextFactory;
 import org.locationtech.spatial4j.shape.Shape;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class GeoNamesQueryLuceneDirectoryIndex extends GeoNamesQueryLuceneIndex {
+
     private static final Logger LOGGER =
             LoggerFactory.getLogger(GeoNamesQueryLuceneDirectoryIndex.class);
 
+    private static final JtsSpatialContextFactory JTS_SPATIAL_CONTEXT_FACTORY =
+            new JtsSpatialContextFactory();
+
+    static {
+        // permits geometry collections with intersecting polygons
+        JTS_SPATIAL_CONTEXT_FACTORY.allowMultiOverlap = true;
+    }
+
+    private static final SpatialContext SPATIAL_CONTEXT =
```

Comment:
Hm, good question. `SpatialContext` states in its javadoc that it is immutable, therefore threadsafe. Since `JtsSpatialContext` extends from that, it would be an API violation for it to be unsafe, and looking at its source code nothing jumps out at me. Are you aware of anything to the contrary?